### PR TITLE
Update dalek version

### PR DESCRIPTION
**Pull request by sakridge**
#### Problem

New dalek version

#### Summary of Changes

Update to 1.0.0-pre3

Fixes #

### CODE REVIEW

File: crate-features/Cargo.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ rand_chacha = { version = "0.1.1" }
 regex-syntax = { version = "0.6.12" }
 reqwest = { version = "0.10.1", default-features = false, features = ["blocking", "rustls-tls", "json"] }
 serde = { version = "1.0.100", features = ["rc"] }
-ed25519-dalek = { version = "=1.0.0-pre.1", features = ["serde"] }
+ed25519-dalek = { version = "=1.0.0-pre.3", features = ["serde"] }
```
Comment (t-nelson) [inline crate-features/Cargo.toml:24]: Do we need to bump `curve25519-dalek` (above, can't comment there) to `"2"` as per the `ed25519-dalek` `Cargo.toml` ?

**Comment by t-nelson** [comment]
@sakridge ready for review when you are!

**Comment by garious** [comment]
@sakridge Before you merge this, can you create a new github issue that points out that all the move dependencies were commented out?

**Comment by sakridge** [comment]
> 
> 
> @sakridge Before you merge this, can you create a new github issue that points out that all the move dependencies were commented out?

https://github.com/solana-labs/solana/issues/9753

File: scripts/cargo-for-all-lock-files.sh [hunk 2]
```diff
@@ -26,7 +26,7 @@ if [[ -n $escape_marker && ${#shifted_args[@]} -gt 0 ]]; then
   done
   shifted_args=()
 else
-  files="$(git ls-files :**Cargo.lock)"
+  files="$(git ls-files :**Cargo.lock | grep -vE 'programs/(librapay|move_loader)')"
```
Comment (ryoqun) [inline scripts/cargo-for-all-lock-files.sh:29]: :eyes:

File: sdk-c/Cargo.toml [hunk 3]
```diff
@@ -16,10 +16,10 @@ crate-type = ["staticlib"]
 bincode = "1.2.1"
 bs58 = "0.3.1"
 libc = "0.2.69"
-rand_chacha = "0.1.1"
-rand_core = { version = ">=0.2, <0.4", default-features = false }
```
Comment (sakridge) [inline sdk-c/Cargo.toml:20]: @jackcmay does it look ok? I'm not sure why it's a range of versions before.

Comment (jackcmay) [reply sdk-c/Cargo.toml:20]: @sakridge Good question, I'm not familiar with why that was done, @jstarry might know

Comment (CriesofCarrots) [reply sdk-c/Cargo.toml:20]: I think this is a question for @TristanDebrunner , actually. I may be talking to him tomorrow, so I'll see if he remembers. That said, I don't think we have any downstream consumers of sdk-c, so I wouldn't be too worried about updating it.

**Pull request merged**